### PR TITLE
Change parameters in cond for surface integrals for older jax versions (power9)

### PR DESCRIPTION
**Pull request by pkim1818**
The surface integrals routine have use the cond function
`    masks = cond(
        has_endpoint_dupe,
        lambda: put(masks, jnp.array([0, -1]), masks[0] | masks[-1]),
        lambda: masks,
    )
`
But in older versions of jax (like the ones used for the traverse builds), cond requires the parameter "operand" which are arguments to the functions. The resulting fix is 

`    masks = cond(
        has_endpoint_dupe,
        lambda _: put(masks, jnp.array([0, -1]), masks[0] | masks[-1]),
        lambda _: masks,
        operand=None
    )
`
Which works since both lambda functions don't take in any arguments. This also works for the newer versions of jax.

### CODE REVIEW

**Review by dpanici — APPROVED** [review]
Soon as all the tests pass, feel free to merge

**Review by ddudt — APPROVED** [review]

**Review by unalmis — APPROVED** [review]

**Review by f0uriest — APPROVED** [review]
